### PR TITLE
Update ACPI crates to edition 2021 and clippy fix

### PR DESCRIPTION
**Pull request by hecatia-elegua**
more like a kitchen sink PR:
- update edition (i.e. adjust imports)
- remove a few static_assertion dependencies which I had already written as `const _: ()` a few weeks ago
- make clippy happy, where the only place of interest is in acpi lib.rs, where instead of using `if let` like clippy wanted I'm instead printing that the case is unhandled
- AcpiTables has an uncomplete sentence in docs, what should we put there?

### CODE REVIEW

File: kernel/acpi/acpi_table/src/lib.rs
Comment (hecatia-elegua) [inline kernel/acpi/acpi_table/src/lib.rs]: ^

File: kernel/acpi/src/lib.rs
Comment (hecatia-elegua) [inline kernel/acpi/src/lib.rs]: ^

File: kernel/acpi/dmar/src/device_scope.rs
Comment (tsoutsman) [inline kernel/acpi/dmar/src/device_scope.rs:22]:
```suggestion
const _: () = assert_eq!(core::mem::size_of::<DeviceScope>(), 6);
const _: () = assert_eq!(core::mem::align_of::<DeviceScope>(), 1);
```

File: kernel/acpi/acpi_table/src/lib.rs
Comment (tsoutsman) [inline kernel/acpi/acpi_table/src/lib.rs]:
```suggestion
///
/// As more ACPI tables are discovered, the single MappedPages object is
/// extended to cover them.
pub struct AcpiTables {
```

File: kernel/acpi/src/lib.rs
Comment (tsoutsman) [reply kernel/acpi/src/lib.rs]: Is this logging statement necessary? I don't think it's an error to encounter an unknown DMAR entry.

**Review by tsoutsman — CHANGES_REQUESTED** [review]
This is great, thank you. Updating Rust editions across the board is on the long list of TODOs.

One small request, could you change the `assert` to `assert_eq` (I commented an example)? This way, if they aren't equal, Rust will tell us the different values rather than just blindly throwing an error.

File: kernel/acpi/acpi_table/src/lib.rs
Comment (kevinaboos) [reply kernel/acpi/acpi_table/src/lib.rs]: oops! lol, thanks for catching that. What @tsoutsman wrote is sufficient.

File: kernel/acpi/src/lib.rs
Comment (kevinaboos) [reply kernel/acpi/src/lib.rs]: agreed, we don't use them yet so let's keep it as is.

**Comment by kevinaboos** [comment]
Thanks, looks good to me! Will merge once the existing comments are addressed.

That `const` assert trick is neat. If you feel so inclined, would you mind submitting another PR that replaces the rest of the places where `static_assertions` is used?

File: kernel/acpi/dmar/src/device_scope.rs
Comment (hecatia-elegua) [reply kernel/acpi/dmar/src/device_scope.rs:22]: I would love to, but currently only `assert!` works in const-context, `assert_eq!` for whatever reason doesn't just call `assert!`. Maybe it has to do with declarative macros and `assert!` being a compiler built-in.
We can revert the change, though since one can hover over e.g. `DeviceScope` here and see size and align with rust-analyzer (in vscode at least), it should not matter that much

File: kernel/acpi/src/lib.rs
Comment (hecatia-elegua) [reply kernel/acpi/src/lib.rs]: warn would have made more sense, but I guess adding an `allow` here does the same thing until we need the other entries

Comment (kevinaboos) [reply kernel/acpi/src/lib.rs]: you could lower it to a `debug` or `trace` if you like, but we really don't need to be logging this since it doesn't provide any useful info atm.

**Comment by hecatia-elegua** [comment]
Done

**Review by tsoutsman — APPROVED** [review]

File: kernel/acpi/src/lib.rs
Comment (kevinaboos) [inline kernel/acpi/src/lib.rs]: might as well change it to the `if let` pattern (similar to what you did in the other file)

**Review by kevinaboos — CHANGES_REQUESTED** [review]
Looks good -- just one minor request. Thanks!

File: kernel/acpi/src/lib.rs
Comment (hecatia-elegua) [reply kernel/acpi/src/lib.rs]: > in acpi lib.rs, where instead of using if let like clippy wanted I'm instead printing that the case is unhandled

I left it like this because I thought a future dev will handle the other variants

**Comment by kevinaboos** [comment]
A recent commit fixed all remaining clippy issues. I think this PR is ready to go once you merge in the latest changes from` theseus_main`.

File: kernel/acpi/src/lib.rs
Comment (kevinaboos) [reply kernel/acpi/src/lib.rs]: Resolved in the main branch already.

**Review by kevinaboos — APPROVED** [review]
Looks good now. Thanks very much!

**Pull request merged**
